components/brave_wallet_ui/utils/enum-utils.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+/**
+ * removes the first character from a Mojo-generated enum key
+ * only if the first character is 'k'
+ */
+export type CleanedFromMojoEnumKey<T extends string | number | symbol> =
+  T extends `${infer U}${infer V}`
+    ? U extends 'k'
+      ? V
+      : Exclude<T, 'MIN_VALUE' | 'MAX_VALUE'>
+    : never
+
+export const getGetCleanedMojoEnumKeys = <T extends object>(mojoEnum: T) => {
```

Comment:
Should we have a `test` for this?

Reply:
tests added: https://github.com/brave/brave-core/compare/ca6f996896732aaf6b0ae9ab899d4f79de5c0683..5e82502821e64824d7567df4cf916945322d2646